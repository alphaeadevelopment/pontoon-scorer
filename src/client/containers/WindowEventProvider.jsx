/* eslint-disable no-undef */
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { debounce } from 'lodash';
import { resize } from '../actions/viewport';

class WindowEventProvider extends Component {
  constructor(props) {
    super(props);
    this.debouncedUpdateSize = debounce(this.updateSize, 100);
  }
  componentDidMount() {
    this.updateSize();
    window.addEventListener('resize', this.debouncedUpdateSize);
  }
  componentWillUnmount() {
    window.removeEventListener('resize', this.debouncedUpdateSize);
  }
  updateSize = () => {
    const { onResize } = this.props;
    onResize({ width: window.screen.availWidth, height: window.screen.availHeight });
  }
  render() {
    return (
      <Fragment>
        {this.props.children}
      </Fragment>
    );
  }
}

export default connect(() => ({}), { onResize: resize })(WindowEventProvider);
