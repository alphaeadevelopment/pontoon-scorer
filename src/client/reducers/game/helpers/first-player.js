import findIndex from 'lodash/findIndex';

export default ({ players, dealer }) => findIndex(players, p => p.idx !== dealer);
