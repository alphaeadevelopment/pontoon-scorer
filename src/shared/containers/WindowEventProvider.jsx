import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import debounce from 'lodash.debounce';
import { setSize } from '../actions/viewport';
import { window } from '../services';

@connect(() => ({}), { setSize })
class WindowEventProvider extends Component {
  constructor(props) {
    super(props);
    this.debouncedUpdateSize = debounce(this.updateSize, 100);
  }
  componentDidMount() {
    this.updateSize();
    window.addEventListener('resize', this.debouncedUpdateSize);
  }
  componentWillUnmount() {
    window.removeEventListener('resize', this.debouncedUpdateSize);
  }
  updateSize = () => {
    const { setSize } = this.props;
    setSize(window.innerWidth, window.innerHeight);
  }
  render() {
    const { children } = this.props;
    return (
      <Fragment>
        {children}
      </Fragment>
    );
  }
}

export default WindowEventProvider;
