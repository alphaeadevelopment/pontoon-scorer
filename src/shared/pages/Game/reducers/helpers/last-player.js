import findLastIndex from 'lodash.findlastindex';

export default ({ players, dealer }) => findLastIndex(players, p => p.idx !== dealer);
