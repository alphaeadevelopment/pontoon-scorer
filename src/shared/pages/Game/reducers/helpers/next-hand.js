export default ({ players, currentPlayer, currentPlayerHand }) => {
  if (players[currentPlayer].hands.length > (currentPlayerHand + 1)) {
    return currentPlayerHand + 1;
  }
  return 0;
};
