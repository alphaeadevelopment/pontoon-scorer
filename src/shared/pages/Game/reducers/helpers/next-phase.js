import { DEALER_HAND, RESULTS } from '../../../../lib/constants/game-phases';

export default ({ phase, players, currentPlayer, currentPlayerHand }) => {
  if (phase === DEALER_HAND) return RESULTS;
  if (players[currentPlayer].hands.length > (currentPlayerHand + 1)) {
    return phase;
  }
  if (currentPlayer + 1 >= players.length) {
    return phase + 1;
  }
  return phase;
};
