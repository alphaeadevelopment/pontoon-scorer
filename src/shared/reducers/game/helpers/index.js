export { default as nextPlayer } from './next-player';
export { default as nextHand } from './next-hand';
export { default as nextPhase } from './next-phase';
export { default as firstPlayer } from './first-player';
export { default as getGamePhaseAfterSetStake } from './get-game-phase-after-set-stake';
export { default as getPlayerAfterSetStake } from './get-player-after-set-stake';

