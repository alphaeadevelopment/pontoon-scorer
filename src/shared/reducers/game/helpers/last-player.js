import findLastIndex from 'lodash/findLastIndex';

export default ({ players, dealer }) => findLastIndex(players, p => p.idx !== dealer);
