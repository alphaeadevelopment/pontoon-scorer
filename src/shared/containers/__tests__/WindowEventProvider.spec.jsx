import React from 'react';
import WindowEventProvider from '../WindowEventProvider';

import { createRender } from '../../../../test/src/helpers/render-helpers';

const render = createRender(WindowEventProvider, { wrapped: true });

describe('<WindowEventProvider />', () => {
  it('renders children', () => {
    const { getTree } = render({
      classes: {},
      children: (
        <p>
          Content
        </p>
      ),
    });

    expect(getTree()).toMatchSnapshot();
  });
});
