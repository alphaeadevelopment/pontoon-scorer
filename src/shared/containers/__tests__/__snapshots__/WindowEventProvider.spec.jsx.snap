// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WindowEventProvider /> renders 1`] = `
<React.Fragment>
  <p>
    Content
  </p>
</React.Fragment>
`;

exports[`<WindowEventProvider /> renders children 1`] = `
<React.Fragment>
  <p>
    Content
  </p>
</React.Fragment>
`;
